### ISSUE TITLE
Check address is valid

### ISSUE THREAD

**Issue by tokyoincode** [Tokyo-Buffalo/pow-wow-calendar#33]
Write a function that will check whether the address is valid or not.

**Comment by tokyoincode** [comment]
@tronicdata We will want to check out the `imports/api/maps/google-maps.js` file. Here's the code block that you might want to hook onto this event. The event fires when the user clicks on a map. You can check the location of the marker that is placed via the `latLng` variable.

```
  eventLocation.addListener('click', function(event) {
      const lat = event.latLng.lat();
      const lng = event.latLng.lng();
      const latLng = {lat: lat, lng: lng};
      marker.setPosition({lat: lat, lng: lng});
      geocodeLatLng(geocoder, map, latLng);
    });
```

**Comment by tronicdata** [comment]
Created a branch for this issue, not sure how to link it to this thread or if it needs to be

**Comment by tronicdata** [comment]
What exactly should we be validating for? Should users be able to choose an address outside of the US? How much validating does the one we just installed on the form cover the address input? All the address validation services I have seen are paid I believe.

**Comment by tokyoincode** [comment]
Sorry, missed the notification.

Form validation will be separate from the google maps marker placement. 

Probably should only work within the USA. We also don't need to use another library. It's either geocoding or reverse-geolocation in the Google docs. 

If we send the coordinates, it should respond with a formatted_address value (JSON)

**Comment by tronicdata** [comment]
When you place a marker it places a formated address in the '.js-map-address' already. Not sure what I need to validate. It wont let me place an address that is in the water. I set the language and region to the US. 

What I do think is that you need to be able to write in a location and when you 'key up' the map needs to update.

**Comment by tronicdata** [comment]
Ok I added in branch for issue -33. Check out the changes I made. Let me know if there is a better way to link the new branch commit inside of this issue thread. 

I added code to validate the location to be in either USA or Canada. If outside those two states then the marker will reset to washington DC. I also added the ability to search for your locale by typing into the input box and on 'focusout' event the map will update and validate the location.

Code is a bit dirty now, took me a few hours to get this far so I will refactor later, let me know what you think!
